command/results_xunit.go

```diff
@@ -131,60 +131,43 @@ func (c *xunitResults) parseAndUploadResults(ctx context.Context, conf *model.Ta
 			return errors.Wrap(err, "error closing xunit file")
 		}
 
-		tests, logs, logIdxToTestIdx = generateLogsForOneFile(testSuites, conf.Task, tests, logs, logIdxToTestIdx)
-	}
-
-	td := client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}
-
-	for i, log := range logs {
-		if ctx.Err() != nil {
-			return errors.New("operation canceled")
-		}
-
-		logId, err := sendJSONLogs(ctx, logger, comm, td, log)
-		if err != nil {
-			logger.Task().Warningf("problem uploading logs for %s", log.Name)
-			continue
-		}
-		tests[logIdxToTestIdx[i]].LogId = logId
-		tests[logIdxToTestIdx[i]].LineNum = 1
+		tests = sendLogsForOneFile(ctx, logger, comm, testSuites, conf.Task, tests, td)
 	}
 
 	return sendJSONResults(ctx, conf, logger, comm, &task.TestResults{tests})
 }
 
-func generateLogsForOneFile(testSuites []testSuite, t *task.Task, tests []task.TestResult, logs []*model.TestLog, logIdxToTestIdx []int) ([]task.TestResult, []*model.TestLog, []int) {
+func sendLogsForOneFile(ctx context.Context, logger client.LoggerProducer, comm client.Communicator,
+	testSuites []testSuite, t *task.Task, tests []task.TestResult, td client.TaskData) []task.TestResult {
```

Comment:
is there some way to break this into different functions (a parsing function and a sending function,) so that we don't need 7 arguments?